Fix for issue 25: Stdin is blocked on Windows

durbanlegend opened the pull request:
See my comment on this issue here: https://github.com/dalance/termbg/issues/25#issuecomment-2420672463
Regards,
Don

dalance commented:
Thank you for your contribution!
I want to merge this PR, there are some questions.

* Unexpected newline
`cargo run` of this PR outputs additional 2 newlines like below.
Do you know why they are shown?

```
Check terminal background color


  Term : Tmux
  Latency: 86.091µs
  Color: detection failed Io { source: Custom { kind: TimedOut, error: "timeout" } }
  Theme: detection failed Io { source: Custom { kind: TimedOut, error: "timeout" } }
```

* Timeout on Windows
\
Why timeout on Windows is changed to the fixed value?
I think 2 seconds is too long.

```rust
let timeout = if cfg!(target_os = "windows") {
    Duration::from_secs(2) // Adjust as needed
} else {
    timeout
};
```

durbanlegend commented:
Thanks for your response! You raise valid points which I've attended to.

Apologies for the added newlines, I didn't even notice them because my focus was elsewhere. I had (unnecessarily) replaced the `write!` of the OSC query with `writeln!`, and for some reason this was inserting not just one but two newlines directly into the output. I've reverted this.

As for the 2 sec Windows timeout override, this arose because I had some timeout issues during the course of development and accepted GPT's suggestion to increase the timeout for Windows. However, I'm seeing latencies of under 500 microseconds and quick responses to OSC 11 from WT 1.22, so I've removed the override.

I also want to apologise for some mildly destructive formatting changes, in particular to the package.metadata.release of `Cargo.toml`. These are not deliberate but are done by `rust-analyzer`, so it's not my intention to impose them.

BTW I've looked further into unit or integration testing for this issue and unfortunately have come to realise that this can't be automated, because `fn from_xterm` (rightly) requires an actual terminal. The closest I can suggest is a sub-crate with a test program that calls `termbg::rgb` and then reads user input from stdin, allowing the user to see if the first character is being swallowed. Something like the following program in my `thag-rs` script runner project. But that might be overkill once this issue is hopefully behind us.
``` rust
/*[toml]
[dependencies]
# Alternate between the following two termbg dependencies to test the difference:
#termbg = "=0.5.2"
termbg = { git = "https://github.com/durbanlegend/termbg" }
*/

use std::io::{self, Read};

fn main() {
    let timeout = std::time::Duration::from_millis(100);

    let _rgb = termbg::rgb(timeout);

    println!("Type in something and see if first character gets swallowed in Windows Terminal");
    let mut buffer = String::new();
    io::stdin().lock().read_to_string(&mut buffer).unwrap();
    println!("buffer={buffer:?}");
}
```

dalance commented:
Thanks! It works fine.
I'll merge this PR and release a new version.

About integrated testing, I think it should be considered in another issue/PR.
